invenio_rdm_records/requests/access/requests.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Access requests for records."""
+
+from flask import current_app, g
+from invenio_access.permissions import authenticated_user, system_identity
+from invenio_i18n import gettext as _
```

Comment:
NOTE: We can't use `lazy_gettext` here because that will create a `flask_babel.speaklater.LazyString` which will cause issues with the JSONSchema.
In contrast, `gettext` will return `str` which is fine.

Reply:
Update: Simplified the fields stored in the DB s.t. translation is unnecessary, and removed the "btw here's the access link" comment after accepting the guest access request.
Only the email is still translated; now with `lazy_gettext`.